### PR TITLE
[airbyte-cdk] Python 3.11 dataclass compatibility

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What

Use dataclass fields with default_factory for 3.11 compatibility. Reimplements https://github.com/airbytehq/airbyte/pull/33798 on top of current master.

## How

Uses `field` with `default_factory` instead of const default values.

## User Impact
None.

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

## TODO
- We might want to do CDK testing on CI, but I'll check local tests and I just want to get this in so folks can run our connectors on 3.11.

### CODE REVIEW

File: airbyte-cdk/python/unit_tests/sources/declarative/parsers/testing_components.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ class TestingSomeComponent(DefaultErrorHandler):
     A basic test class with various field permutations used to test manifests with custom components
     """
 
-    subcomponent_field_with_hint: DpathExtractor = DpathExtractor(field_path=[], config={}, parameters={})
+    subcomponent_field_with_hint: DpathExtractor = field( default_factory=lambda: DpathExtractor(field_path=[], config={}, parameters={}))
```
Comment (natikgadzhi) [inline airbyte-cdk/python/unit_tests/sources/declarative/parsers/testing_components.py:21]:
```suggestion
    subcomponent_field_with_hint: DpathExtractor = field(default_factory=lambda: DpathExtractor(field_path=[], config={}, parameters={}))
```

**Review by girarda — APPROVED** [review]
lgtm ✅

**Comment by aaronsteers** [comment]
Nice!

Thanks, @natikgadzhi ! 👏

**Comment by natikgadzhi** [comment]
Heads up, if you're here because your connector of choice does not work on 3.11, it could be that it's using an outdated CDK. Here's a PR that maybe fixes Prestashop: https://github.com/airbytehq/airbyte/pull/41656

**Pull request merged**